src/main/java/com/exasol/adapter/dynamodb/DynamodbAdapter.java

```diff
@@ -31,51 +37,41 @@
+			final SchemaMetadata schemaMetadata = SchemaMappingDefinitionToSchemaMetadataConverter
+					.convert(schemaMappingDefinition);
+			return CreateVirtualSchemaResponse.builder().schemaMetadata(schemaMetadata).build();
+		} catch (final IOException e) {
+			throw new AdapterException("Unable create Virtual Schema \"" + request.getVirtualSchemaName()
+					+ "\". Cause: \"" + e.getMessage(), e);// NOSONAR
+		}
+	}
+
+	private SchemaMappingDefinition getSchemaMappingDefinition(final AdapterRequest request)
+			throws AdapterException, IOException {
+		final AdapterProperties adapterProperties = new AdapterProperties(
+				request.getSchemaMetadataInfo().getProperties());
+		final DynamodbAdapterProperties dynamodbAdapterProperties = new DynamodbAdapterProperties(adapterProperties);
+		final File path = dynamodbAdapterProperties.getMappingDefinition();
+		if (!path.exists()) {
+			throw new AdapterException(String.format("the specified mapping file (%s) could not be found.", path));
+		}
+		final MappingFactory mappingFactory = new JsonMappingFactory(path);
+		return mappingFactory.getSchemaMapping();
 	}
 
 	/**
 	 * Creates a connection to DynamoDB using the connection details set in
 	 * {@code CREATE CONNECTION}.
 	 */
-	private DynamoDB getConnection(final ExaMetadata exaMetadata, final AbstractAdapterRequest request)
+	private AmazonDynamoDB getConnection(final ExaMetadata exaMetadata, final AbstractAdapterRequest request)
 			throws ExaConnectionAccessException {
 		final AdapterProperties properties = getPropertiesFromRequest(request);
 		final ExaConnectionInformation connection = exaMetadata.getConnection(properties.getConnectionName());
-		return getDynamodbConnection(connection.getAddress(), connection.getUser(), connection.getPassword());
-	}
-
-	/**
-	 * Creates a DynamoDB client for a given uri, user and key.
-	 * 
-	 * @param uri
-	 *            either aws:<REGION> or address of local DynamoDB server (e.g.
-	 *            http://localhost:8000)
-	 * @param user
-	 *            aws credential id
-	 * @param key
-	 *            aws credential key
-	 * @return DynamoDB client
-	 */
-	protected static DynamoDB getDynamodbConnection(final String uri, final String user, final String key) {
-		final String AWS_PREFIX = "aws:";
-		final String AWS_LOCAL_REGION = "eu-central-1";
-		final BasicAWSCredentials awsCredentials = new BasicAWSCredentials(user, key);
-		final AmazonDynamoDBClientBuilder clientBuilder = AmazonDynamoDBClientBuilder.standard()
-				.withCredentials(new AWSStaticCredentialsProvider(awsCredentials));
-		if (uri.startsWith(AWS_PREFIX)) {
-			clientBuilder.withRegion(uri.replace(AWS_PREFIX, ""));
-		} else {
-			clientBuilder.withEndpointConfiguration(new AwsClientBuilder.EndpointConfiguration(uri, AWS_LOCAL_REGION));
-		}
-		return new DynamoDB(clientBuilder.build());
+		return DynamodbConnectionUtil.getLowLevelConnection(connection.getAddress(), connection.getUser(),
```

Comment:
Calling something `...Util` should always be an alarm signal.
Better create a class `DynamoDbConnectionBuilder`. Also avoid static code unless strictly necessary.

Reply:
i think Factory fits better here.

Reply:
Agreed.